Public/Src/Utilities/KeyValueStore/KeyValueStoreAccessor.cs

```diff
@@ -638,17 +736,75 @@ public Possible<Unit> Use(Action<IBuildXLKeyValueStore> use)
                     return Unit.Void;
                 }, use);
         }
+        
+        private (bool lockdown, bool rethrow) StrictErrorPolicy(Failure<Exception> failure)
+        {
+            Contract.Requires(failure != null);
+
+            var ex = failure.Content;
+            var isUserError = true;
+            if (ex is RocksDbSharpException || ex is System.Runtime.InteropServices.SEHException)
+            {
+                // The SEHException class handles SEH (structured exception handling) errors that are thrown from 
+                // unmanaged code, but that have not been mapped to another .NET Framework exception. The SEHException
+                // class also corresponds to the HRESULT E_FAIL (0x80004005).
+                isUserError = false;
+            }
+
+            m_failureHandler?.Invoke(failure);
+            return (true, isUserError);
+        }
+
+        private (bool lockdown, bool rethrow) RelaxedErrorPolicy(Failure<Exception> failure)

```

Comment:
When we discuss this thing, I suggested (I think) to actually move the entire logic into the clients.

For instance, instead of having strict vs relaxed mode, I would delegate this decision to the client in a form of a call back.

It means that instead of taking two argumetns (the mode and the handler), the clients will be able to provide a handler similr to one from TaskUnbservedException handler when the callback can set the result whether the error is handled (non strict mode) or not (strict mode).

// the client code:

var accessor = new Accessor(MyHandler);

private void Handler(HandlerArgs args) {
  // the client is strict:
  args.Handled = true;
}